### PR TITLE
Add support for emotes via `emote_mapper`

### PR DESCRIPTION
**Pull request by ModProg**
I'm not sure if this makes sense to have in here.

I use it for my self, but as you want to implement Emotes via different means, this might be redundant, as using [emote_mapper](https://lib.rs/crates/emote_mapper) is not very ergonomic.

You need to create a font with [twitchmote](https://github.com/ModProg/twitchmote), install the font file and configure twt to use it.

### CODE REVIEW

**Comment by Xithrius** [comment]
Will review at a later time.

File: src/handlers/app.rs [hunk 1]
```diff
@@ -1,5 +1,10 @@
-use std::collections::{HashMap, VecDeque};
+use std::{
+    collections::{HashMap, VecDeque},
+    path::PathBuf,
```
Comment (Xithrius) [inline src/handlers/app.rs:3]: This is never used.

File: src/handlers/app.rs [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+};
```
Comment (Xithrius) [inline src/handlers/app.rs:4]: `path::PathBuf` is never used.

```suggestion
use std::collections::{HashMap, VecDeque};
```

**Comment by Xithrius** [comment]
I feel like there should be a more consistent way of implementing this, instead of relying on the user for resources.

**Comment by ModProg** [comment]
> I feel like there should be a more consistent way of implementing this, instead of relying on the user for resources.

yes, but that would only work with some image in terminal protokol like  sixel. I wrote this for someone who uses Alacrity which does not have that. 

But yeah, I would agree that this is the PR with the least reason to be merged.

**Comment by Xithrius** [comment]
> I would agree that this is the PR with the least reason to be merged.

Then, shouldn't I close this?

**Comment by Xithrius** [comment]
Although since this does work, maybe this be stored on a branch until a more consistent way is found.

I know that [anthonywritescode](https://www.twitch.tv/anthonywritescode) has emotes in his twitch terminal chat, but I'm yet to find the repository.

**Comment by ModProg** [comment]
yeah, this can be understood as a fallback for people not willing to use a terminal with support for an image protocol like sixel or kitty.

Due to the limited number of possible Unicode code points (thank utf-16 compatibility for that), it is impossible to provide a font with all Twitchemotes, so the user would always need to choose the emotes they want them self, and we cannot provide them (maybe this can be worked around with ligatures, but alacrity does not support those either).

I use a terminal with sixel, iterm and kitty support, so I will be perfectly content with a more ergonomic solution using one of these.

**Comment by Xithrius** [comment]
If we were to go for the more ergonomic solution for supported users and have the current state of this PR as a backup for others, what was be used in rust to approach this?

**Comment by ModProg** [comment]
well we could make it feature based (but that is maybe not the greatest thing for bins IMO) or we just have both implemented switchable via config (we could also detect terminal support but idk if that is necessary).

We could include support for the font generation via the cli, but I would have to revisit that project probably and have a look what the actual dependencies are IIRC there was some heavy use of python currently, so this is maybe not something you'd want included by default.

I also expect the kind of people who use a terminal chat and are not willing to run it in a terminal with image support would be able to handle building the font through a seperate project and maybe also building twitch-tui with a feature flag. 

feel free to just keep this here and revisit it after figuring out the image based emots.

**Comment by ModProg** [comment]
> If we were to go for the more ergonomic solution for supported users and have the current state of this PR as a backup for others, what was be used in rust to approach this?

oh and I think you can install from a branch with cargo IIRC if that is what you meant

**Comment by Xithrius** [comment]
Both answers were what I was looking for. I'll leave this PR here for later. I shall be focusing on getting 2.0 done.

**Comment by Xithrius** [comment]
I will be closing this PR for now due to the large amounts of changes that this program is going under.

If I decide that this feature shall be supported, I'll re-open this PR (it may actually be best for it to be in a new PR depending on the conflicts in the future).

**Comment by ModProg** [comment]
Makes sense exspecially with the background of this not being the prefered way to implement it.

**Comment by Xithrius** [comment]
Agreed.

**Pull request closed without merging**